madmom/ml/crf.py

```diff
@@ -0,0 +1,76 @@
+# encoding: utf-8
+"""
+This module contains an implementation of Conditional Random Fields (CRFs)
+"""
+import numpy as np
+
+from ..processors import Processor
+
+
+class ConditionalRandomField(Processor):
+    """
+    Implements a linear-chain Conditional Random Field using a
+    matrix-based definition:
+
+    P(Y|X) = exp(E(Y,X)) / Σ_Y'[E(Y', X)]
+    E(Y,X) = Σ_{i=1..N} [ y_{n-1}.T * A * y_n + y_n.T * c + x_n.T * W * y_n ] +
+             y_0.T * π + y_N.T * τ
+
+    Parameters:
+    -----------
+    pi : numpy array
+        Initial potential of the CRF. Also defines the number of states.
+    tau : numpy array
+        Potential of the last variable of the CRF.
+    c : numpy array
+        Label bias potential
+    A : numpy array
+        Matrix defining the transition potentials, where the rows are the
+        'from' dimension, and columns the 'to' dimension.
+    W : numpy array
+        Matrix defining the observation potentials, where the rows are the
+        'observation' dimension, and columns the 'state' dimension
+    """
+
+    def __init__(self, pi, tau, c, A, W):
+        self.pi = pi
+        self.tau = tau
+        self.c = c
+        self.A = A
+        self.W = W
```

Comment:
Argument names should be lowercase (PEP8), but if those are always called `A` and `W` I'd leave them as they are.